crates/esplora/tests/async_ext.rs

```diff
@@ -109,6 +109,27 @@ pub async fn test_update_tx_graph_without_keychain() -> anyhow::Result<()> {
         )
         .await?;
 
+    // Check to see if we have the floating txouts available from our two created transactions'
+    // previous outputs in order to calculate transaction fees.
+    graph_update.full_txs().for_each(|tx| {
+        let calculate_fee_result = graph_update.calculate_fee(tx.tx);
+        let get_tx_result = env.bitcoind.client.get_transaction(&tx.txid, None);
+
+        // Check if fee calculation and transaction retrieval were successful
+        let tx_fee_exists = get_tx_result.as_ref().map_or(false, |tx| tx.fee.is_some());
+        assert_eq!(calculate_fee_result.is_ok(), tx_fee_exists);
+
+        // Check if the calculated fee value matches the fee from transaction retrieval
+        if let (Ok(fee), Ok(transaction)) = (calculate_fee_result, get_tx_result) {
+            let tx_fee = transaction
+                .fee
+                .unwrap_or_else(|| panic!("Fee not found for transaction {:?}", tx.txid))
```

Comment:
You are sure that `transaction.fee` exists, as at this point `calculate_fee_result` is ok, and you check above that `tx_fee_exist` is equal to `calculate_fee_result.is_ok()`. This can just be an `unwrap` or `expect(*explanation of why this must exist*)`